### ISSUE TITLE
`entt::registry::clear()` does not "clear enough"

### ISSUE THREAD

**Issue by skaarj1989** [skypjack/entt#1106]
Minimal repro:
```cpp
entt::registry r;
r.create();
// ...
r.clear();
entt::snapshot_loader loader{ r };
```
https://github.com/skypjack/entt/blob/fedcb920ce0068c35ffbc66fd4e84864e6ef71ef/src/entt/entity/snapshot.hpp#L167

**Comment by skypjack** [comment]
Ahaha love the title! 😂 
In fact, the problem isn't in the `clear` function (which stays true to the same behavior it had in the previous versions).
The problem is the check in the constructor of the snapshot loader, which is more stringent then before instead (and somewhat wrong too).

**Comment by skypjack** [comment]
Fix available on branches `wip` and `v3.13.x` (next patch release for version 3.13).
Let me know if it doesn't work for you. Thanks. 👍
